### ISSUE TITLE
When Aspect Ratio = 1, width and height are different

### ISSUE THREAD

**Issue by ronflima** [jamesssooi/Croppr.js#20]
Options used:

- aspectRatio: 1
- onCropEnd: simple function that simply prints on the console the _data_ parameter

When aspectRatio is 1, it is expected to have a cropping region that is a perfect square, i.e., width must be equal to height. 

I have found a difference. width is slightly bigger than the height. The difference varies from 1 to 3. U have ran some scenarios and found the following data:

```javascript
{x: 653, y: 2550, width: 1474, height: 1473} 
{x: 1942, y: 1094, width: 1474, height: 1473}
{x: 1942, y: 1009, width: 1643, height: 1642}
{x: 1942, y: 881, width: 1901, height: 1899}
{x: 1942, y: 881, width: 2333, height: 2330}
{x: 4199, y: 3211, width: 27, height: 27}
{x: 2430, y: 1488, width: 2329, height: 2327}
{x: 901, y: 662, width: 2329, height: 2327} 
{x: 3621, y: 2207, width: 596, height: 595}
```
Also important to mention: the height is never the height of the picture - in my scenarios I'm using a rectangular picture of 5334x4032 pixels. Maximum height I get is always 5 pixels less even if I enlarge the cropping area as much as possible. 

I will run some extra scenarios in order to determine correctly why this is happening.

**Comment by jamesssooi** [comment]
Hi @ronflima, I can confirm that I can replicate this issue on my end. Seems to happen on any source pictures that are not square in itself. Should be a simple rounding error, will dig deeper to find out!

**Comment by jamesssooi** [comment]
Fixed in [v2.3.1](https://github.com/jamesssooi/Croppr.js/releases/tag/v2.3.1)!

The problem was with the `Element.offsetWidth` and `Element.offsetHeight` properties, which is relied upon by Croppr to calculate the final crop value. The default behaviour of these properties is to return a **rounded** value, which is of course — the cause of the inaccuracy in the final reported value.

Replacing these with `getBoundingClientRect()`, which returns the actual sizes of the element without rounding fixes this issue completely.

Thanks @ronflima for bringing this up! It was a pretty major bug that wasn't caught via testing.
